File: Solutions/1217_ Minimum _Cost_to_Move_Chips_to_The_Same_Position/1217_ Minimum _Cost_to_Move_Chips_to_The_Same_Position.cs
public class Solution {
    public int MinCostToMoveChips(int[] position) {
        int even = 0, odd = 0;
        
        foreach(var p in position){
            if(p%2 == 0)
                even++;
            else
                odd++;
        }
        
        return Math.Min(even, odd);
    }
}

//Second fastest
public class Solution {
    public int MinCostToMoveChips(int[] position) {
        int even =0;
        int odd = 0;
        for(int i=0;i<position.Length;i++)
        {
            if(position[i]%2 == 0)
                even++;
            else
                odd++;
        }
        return Math.Min(even, odd);
    }
}

//Fastest
public class Solution {
    public int MinCostToMoveChips(int[] position) {
        
        int odd = 0;
        int even = 0;
        
        int[] pos = new int[101];
        
        for(int i=0;i<position.Length;i++)
        {
            if(position[i] %2 == 0) even++;
            else odd++;
        }

        return Math.Min(odd,even);
        
    }
}

File: Solutions/1217_ Minimum _Cost_to_Move_Chips_to_The_Same_Position/1217_ Minimum _Cost_to_Move_Chips_to_The_Same_Position.js
var minCostToMoveChips = function(position) {
    return Math.min(position.reduce((count, pos) => count + (pos & 1), 0), position.reduce((count, pos) => count + 1 - (pos & 1), 0));
};

//Third fastest
/**
 * @param {number[]} position
 * @return {number}
 */
var minCostToMoveChips = function(position) {
    let even = position.filter(e => e % 2 === 0).length;
    let odd = position.length - even;
    return Math.min(even, odd)
        
};

//Second fastest
/**
 * @param {number[]} position
 * @return {number}
 */
var minCostToMoveChips = function(position) {
    let evenCoinsSum = 0;
    let oddCoinsSum = 0;
    for (let i = 0; i < position.length; i++) {
        if (position[i] % 2 == 0) {
            evenCoinsSum++;    
        } else {
            oddCoinsSum++;    
        }
    }    
    return Math.min(evenCoinsSum, oddCoinsSum);
};

//Fastest
/**
 * @param {number[]} position
 * @return {number}
 */
var minCostToMoveChips = function(position) {
    let evenCoinsSum = 0;
    let oddCoinsSum = 0;
    for (let i = 0; i < position.length; i++) {
        if (position[i] % 2 == 0) {
            evenCoinsSum++;    
        } else {
            oddCoinsSum++;    
        }
    }    
    return Math.min(evenCoinsSum, oddCoinsSum);
};